man/adoc/bpftrace.adoc

```diff
@@ -3132,6 +3132,41 @@ tracepoint:syscalls:sys_enter_execve {
 }
 ----
 
+[#functions-task_from_pid]
+=== task_from_pid
+
+.variants
+* `struct task_struct *task_from_pid(pid)`
+
+*compile time*
+
+Get 'struct task_struct *' pointer from pid.
+
+`task_from_pid()` needs to be used together with `task_release()`, otherwise it will fail the verifier check.
```

Comment:
Currently, our general strategy is to minimize the possibilities to trigger verifier errors from bpftrace scripts as those may be quite hard to debug for users.

So, I want to avoid merging this in the current form. However, there's clearly value in exposing these (and other ptr-acquiring) kfuncs so let's discuss how we do it in #4198.

Reply:
Thanks, If we have a conclusion, I still want to do the work of `task_from_pid`, because I want to use this kfunc, thanks again.

Reply:
If we changed the `Variable` types in codegen to a `ScopedExpr`-style object, we could allow for the possibility of attached a "destructor" method for when those are reassigned to a new value or the scope ends. (Codegen would need to explicitly flush all of these at the end of the probe also, to ensure these are run.)

The resulting expression of this function call could carry around its own destructor, which would be the call to the release function. WDYT?